client/my-sites/domains/domain-management/settings/cards/domain-forwarding-card.tsx

```diff
@@ -113,12 +114,27 @@ export default function DomainForwardingCard( { domain }: { domain: ResponseDoma
 				setProtocol( forwarding.isSecure ? 'https' : 'http' );
 				setIsPermanent( forwarding.isPermanent );
 				setForwardPaths( forwarding.forwardPaths );
+				setSubdomain( forwarding.domain );
 			}
 		} catch ( e ) {
 			// ignore
 		}
 	}, [ isLoading, forwarding, setTargetUrl, setProtocol ] );
 
+	const handleDomainOriginChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
+		setSubdomain( withoutHttp( event.target.value ) );
+
+		const CAPTURE_SUBDOMAIN_RGX = /^[a-zA-Z0-9-]{0,}$/i;
```

Comment:
Would that mean that the user can not set a subdomain as `foo.bar`, for example so they can set a redirect for `foo.bar.example.com`

Reply:
In the current version we didn't expect to allow subdomains like `foo.bar`.mydomain.com. Is this an usual user's need? I'm happy to add support to that if needed.

Reply:
I think we also want to limit the length of the part to 63 characters by changing `{0,}` to `{0,63}` We also need to add this constraint to the back end.

> Each element of a domain name separated by [.] is called a “label.” The maximum length of each label is 63 characters, and a full domain name can have a maximum of 253 characters.

source: https://www.nic.ad.jp/timeline/en/20th/appendix1.html